### DISCUSSION TITLE
SDT-3D view

### DISCUSSION THREAD

**Discussion by PengfeiM** [coreemu/core#543] [Q&A]
Hello again,
I would like to know if I can use a python script to change the nodes position in run time and display the nodes' location and the status(if they can communicate with each other such as a simple ping) of links between nodes in sdt-3d.
Or if I can use a python scripts to change the nodes position in run time and record the  nodes' location and the status(if they can communicate with each other such as a simple ping) of links between nodes.
**In summary**, there are functions bellow I'd like to know if it is supported:
* change the nodes' position in run time using python scripts
* record the nodes' position and the links' status between nodes
* display the nodes' position and the links' status between nodes in sdt-3d in run time.

**Comment by bharnden** [comment]
There are multiple ways to move nodes, gRPC API, `coresendmsg`, `core-cli`, or if you are driving the whole scenario by way of python you can do it in code as well.

If you are driving the scenario then you would have some idea of what link status means, but then that depends on what you are looking for. You can listen for link updates in gRPC from an event listener, but that would really only be for wireless links or listening to some other script changing things.

CORE in general sets things up to run and doesnt really do anything more after it is started, so if you are looking for data collection about what is going in within the nodes, that would be up to you to do.

As far as SDT goes, CORE has basic integration when you turn the SDT session option on and setup the right address and port, assuming SDT is already running. Then when you start a scenario in the GUI or code it would send node positions and links to SDT.

Alternatively, you could interact with SDT yourself, using there API defined in this (document)[https://github.com/USNavalResearchLaboratory/sdt/blob/master/doc/sdt.pdf]. SDT is someone elses tool, so if you want to know more, I would reach out to them.

**Comment by PengfeiM** [comment]
So, I wonder if there is any interface or configuration that can be changed to change the location of the nodes in sdt3d. 
I read some of the codes in core and find out that I should change the parameter: "refgeo". 
But I can not be sure about where to change that parameter. 
Your help will be appreciated.
